tests/unit_tests/vecpar/rk_stepper_vecpar.hpp

```diff
@@ -0,0 +1,85 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#if defined(array)
```

Comment:
If this block appears in several areas of the code, should we not have a central place in `detray/plugin/algebra/definitions.hpp` which runs the if/else clause ?

Reply:
Sounds like a good idea! I will create a task and try to clean it up!

Reply:
Centralising this bad design should help with getting rid of it completely, so I'm not against that. But just to remind people, in the end we will need to stop relying on compiler defitions for choosing an algebra on the source file level. We will need to make the choice object-by-object, using templating.

Reply:
I created issue #240 . @krasznaa I'm happy to implement it the way you think it is better.